CMakeLists.txt

```
@@ -0,0 +1,96 @@
+# Minimum cmake version required by scikit-build-core: 3.15
+# Minimum cmake version required by `Python_SOABI`: 3.17
+cmake_minimum_required(VERSION 3.17)
+
+set(CLP_FFI_PY_INSTALL_LIBS ON CACHE BOOL "Whether to install the built libraries." FORCE)
+
+if (DEFINED SKBUILD_PROJECT_NAME)
+    set(CLP_FFI_PY ${SKBUILD_PROJECT_NAME} CACHE STRING
+            "Use scikit-build-core's project name." FORCE)
+    set(CLP_FFI_PY_VERSION ${SKBUILD_PROJECT_VERSION} CACHE STRING
+            "Use scikit-build-core's project version." FORCE)
+else()
+    # The project is configured outside of scikit-build-core. Normally this is for build testing
+    # purpose, or loading the project into CLion. We don't install the built library in this config.
+    message(WARNING "Configured outside of scikit-build-core. Library installation is disabled.")
+    set(CLP_FFI_PY_INSTALL_LIBS OFF CACHE BOOL
+            "Disable library installation when building outside of scikit-build-core." FORCE)
+    set(CLP_FFI_PY "clp-ffi-py-local-build" CACHE STRING "Use local build name." FORCE)
+    set(CLP_FFI_PY_VERSION "0.0.12" CACHE STRING "Use local build version." FORCE)

```

we probably don't need a specific version number here / don't want to update those in the future since we're specifying those fallbacks only for IDE setups?
```suggestion
    set(CLP_FFI_PY_VERSION "0.0.0" CACHE STRING "Use local build version." FORCE)
```